test/upgradeable/behavior/catalog.ts

```diff
@@ -0,0 +1,286 @@
+import { ethers, upgrades } from 'hardhat';
+import { expect } from 'chai';
+import { Contract } from 'ethers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { IOtherInterface, IERC165, IRMRKCatalog } from '../../interfaces';
+
+async function shouldBehaveLikeCatalog(contractName: string, metadataURI: string, type: string) {
```

Comment:
We should have this behavior receive the contract, so we do not need 2 versions.